### ISSUE TITLE
serial.tools.list_ports.grep() can't find my device (Windows)

### ISSUE THREAD

**Issue by BrendanSimon** [pyserial/pyserial#365]
I have a USB device (based on STM32F4 microcontroller) with VID, PID and Product Description setup in the USB descriptors.

```
VID = 0x0483
PID = 0x5740
Product Description = "ACME thing-a-me-bob"
```

On macOS, using `serial.tools.list_ports.grep()` works great to locate my device.  e.g. `p = grep("thing-a-me-bob")` or `p = grep("ACME")

However on Windows (Win 10), grep does not find my device.  If i use `grep("")` then it does list my device (along with all the others) but it just shows:

`COM3 - USB Serial Device (COM3)`

The device is just using the standard Microsoft CDC driver that comes with Win 10.  I would have thought however that simple/standard thinks like the product description string would be read from a a USB device and used (just like the standard macOS driver does).

Is there a way to get `serial.tools.list_ports.grep()` to work with standard windows drivers?

Note: `grep("0483:5740") does find my device, however this a reasonably commonly used set of VID:PID values, provided by ST Micro.

Note: I've also asked the same question on [StackOverflow](https://stackoverflow.com/questions/51345594/pyserial-cant-find-device-using-serial-tools-list-ports-grep) (if that's a more appropriate place for this issue) ?

**Comment by HazCod** [comment]
Additionally, what devices are supported on Win10? What should I look out for?

**Comment by BrendanSimon** [comment]
I'm guessing any device that shows up as a "COM" port in Device Manager.
i.e. a VCP or CDC device, which can be opened with any program that read/write a byte stream.

**Comment by zsquareplusc** [comment]
The issue here is, that Windows uses the descriptions provided by the driver (`.inf` files) and not the values from the USB descriptor. Under e.g. Win7 manufacturers needed to provide an `.inf` file even if the built-in `USBSER.SYS` driver was loaded and therefore could specify their own name. Now with Win10, there is a generic driver that detects the CDC/ACM device class and the built-in `.inf` file has that generic name hard coded.

The best workaround is to search for VID/PID instead :/  You could also load that specific `.inf` file to get the old description back.

`list_ports_windows.py` uses the `SetupDi` API to get the device info. If there were a "registry property" that contained the strings from the real device's USB descriptor, then that could be added. But I do not think that this information is available... see [SetupDiGetDeviceRegistryPropertyW](
https://docs.microsoft.com/en-us/windows/desktop/api/setupapi/nf-setupapi-setupdigetdeviceregistrypropertyw) for a list of available data.

**Comment by BrendanSimon** [comment]
Yes, I ended up just searching for the VID/PID and then trying to talk to my device.  If it responded great, else try the next one.  A bit of a faff but it works.
